torch_agents/cogment_verse_torch_agents/hf_sb3/sb3_adapter.py

```diff
@@ -0,0 +1,152 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import copy
+import logging
+from collections import namedtuple
+
+import cogment
+
+import torch
+from cogment.api.common_pb2 import TrialState
+from cogment_verse import AgentAdapter
+
+# from cogment_verse import MlflowExperimentTracker
+from cogment_verse_torch_agents.utils.tensors import tensor_from_cog_obs
+from data_pb2 import (
+    ActorParams,
+    AgentAction,
+    AgentConfig,
+    EnvironmentConfig,
+    EnvironmentParams,
+    EnvironmentSpecs,
+    TrialConfig,
+    SimpleSB3TrainingRunConfig,
+)
+from huggingface_sb3 import load_from_hub
+from stable_baselines3 import PPO
+
+# from stable_baselines3.common.evaluation import evaluate_policy
+
+SimpleSB3Model = namedtuple("SimpleSB3Model", ["model_id", "version_number", "policy_network"])
+
+log = logging.getLogger(__name__)
+
+# pylint: disable=arguments-differ
+class SimpleSB3AgentAdapter(AgentAdapter):
+    def __init__(self):
+        super().__init__()
+        self._dtype = torch.float
+
+    @staticmethod
+    async def run_async(func, *args):
+        """Run a given function asynchronously in the default thread pool"""
+        event_loop = asyncio.get_running_loop()
+        return await event_loop.run_in_executor(None, func, *args)
+
+    def _create_actor_implementations(self):
+        async def impl(actor_session):
+            actor_session.start()
+
+            # config = actor_session.config
+
+            checkpoint = load_from_hub(
+                repo_id="ThomasSimonini/ppo-LunarLander-v2",
+                filename="ppo-LunarLander-v2.zip",
+            )
+            model = PPO.load(checkpoint)
+
+            @torch.no_grad()
+            def compute_action(event):
+                obs = tensor_from_cog_obs(event.observation.snapshot, dtype=self._dtype)
+                obs = torch.unsqueeze(obs, dim=0)
+                action = model.predict(obs)
+
+                return action
+
+            async for event in actor_session.event_loop():
+                if event.observation and event.type == cogment.EventType.ACTIVE:
+                    action = await self.run_async(compute_action, event)
+                    actor_session.do_action(AgentAction(discrete_action=action[0]))
+
+        return {
+            "simple_sb3": (impl, ["agent"]),
+        }
+
+    def _create_run_implementations(self):
+        async def sample_producer_impl(run_sample_producer_session):
+            assert run_sample_producer_session.count_actors() == 1
+
+            async for sample in run_sample_producer_session.get_all_samples():
+                if sample.get_trial_state() == TrialState.ENDED:
+                    break
+
+                log.info("Got raw sample")
+
+        async def run_impl(run_session):
```

Comment:
I don't think you need that, you should be able to use the `play` implementation.